src/common/utils2.cpp

```diff
@@ -0,0 +1,69 @@
+#include "utils2.hpp"
+
+#include <chrono>
+#include <string>
+#include <algorithm>
+#include <iostream>
+#include <numeric> //iota
+
+
+using Clock = std::chrono::steady_clock;
+using std::chrono::time_point;
+using std::chrono::duration_cast;
+using std::chrono::microseconds;
+
+struct cScopeTimer::sPimpl {
+    time_point<Clock> start;
+    time_point<Clock> end;
+    
+    sPimpl()
+    {
+        start = Clock::now();
+    }
+    
+    ~sPimpl(){
+        end = Clock::now();
+        microseconds diff = duration_cast<microseconds>(end - start);
+        std::cout << " took=" << diff.count() << "ms !\n";
+    }
+};
+
+cScopeTimer::cScopeTimer() 
+	: aPimpl(new sPimpl())
+{}
+
+/**
+ * Calculates the Levenshtein distance of two strings.
+ * @author http://en.wikibooks.org/wiki/Algorithm_Implementation/Strings/Levenshtein_distance#C.2B.2B
+ * comparison test was done here http://cpp.sh/2o7w
+ */
+int levenshtein(const std::string &s1, const std::string &s2)
```

Comment:
Telling me to not use "auto" everywhere and reading your code afterwards is a bit strange O.o